README.md

```diff
@@ -1,114 +1,103 @@
 
-* REGION
+- Instead of checking non-deleted tenants' timelines instantly, the tool attempts to create separate tasks (futures) for that,
+  complicating the logic and slowing down the process, this should be fixed and done in one "task".
 
-A region where the bucket is located at.
+- The tool does uses only publicly available remote resources (S3, console) and does not access pageserver/safekeeper nodes themselves.
+  Yet, its S3 set up should be prepared for running on any pageserver/safekeeper node, using node's S3 credentials, so the node API access logic could be implemented relatively simply on top.
 
-* BUCKET
+- Tool has `copied_definitions` module that contains copy-pasted storage primitives with parsing and deserializing definitions for them — that might get out of date with future storage development and should be solved either by moving the storage definitions into a separate crate (then it can be referenced via git url already) or unifying the clenup project with the storage (unclear value: extra infrastructure metadata exposure + garbage tenants and timelines in storage is a bug and should not appear normally, the work on this is ongoing: https://github.com/neondatabase/neon/issues/3889)
+  Similarly S3 paths for safekeeper and pageserver are hardcoded, but not likely the structure changes in the near future.
 
-Bucket name
+### Future concerns
 
-* CLOUD_ADMIN_API_URL
+If the tool is left as is, a number of things might break:
 
-The URL base to use for checking tenant/timeline for existence via the Cloud API.
+- `copied_definitions` might get out of date and become incompatible with the future format changes: do update these regularly or library-ify/merge the codebases
 
-* CLOUD_ADMIN_API_TOKEN
+- Console Admin API seems to be using "v1" version? V2 is already in the wild and the old version might be obsoleted eventually
 
-The token to provide when querying the admin API.
-Get one on the corresponding console page, e.g. https://console.stage.neon.tech/app/settings/api-keys
+- S3 format will change for paths also or the layer creation/deletion logic might change on pageserver: if tool does not get obsolete, it needs to get adapted for this
 
-* NODE_KIND: safekeeper/pageserver
+## Cleanup procedure:
 
-Which S3 key paths to construct when checking for S3 objects' existence.
+### Pageserver preparations
 
-* TRAVERSING_DEPTH: tenant/timeline, default: tenant
+If S3 state is altered first manually, pageserver in-memory state will contain wrong data about S3 state, and tenants/timelines may get recreated on S3 (due to any layer upload due to compaction, pageserver restart, etc.)
 
-Whether to check and delete tenants only (by default), or also check non-deleted tenants' timelines for existence too.
-For pageserver, this would also include index_part.json validation for all non-deleted timelines, not only timeline existence check.
+So first, we need to remove deleted in console tenants/timelines from pageservers.
```

Comment:
deleted files* (or layers)

Reply:
I think this was pre-existing text, it meant something like "We need to remove tenants/timelines which are deleted in the console from the pageservers' local storage".  I've updated it.  6e0059d0be71507817416ead4c31adfe6c835032